packages/block-editor/src/components/inserter/quick-inserter.js

```diff
@@ -154,26 +154,22 @@ export default function QuickInserter( {
 		[ filterValue, patterns ]
 	);
 
-	const setInserterIsOpened = useSelect(
-		( select ) =>
-			select( 'core/block-editor' ).getSettings()
+	const { setInserterIsOpened, blockIndex } = useSelect( ( select ) => {
+		const { getSettings, getBlockIndex } = select( 'core/block-editor' );
+		return {
+			setInserterIsOpened: getSettings()
 				.__experimentalSetIsInserterOpened,
-		[]
-	);
-
-	const previousBlockClientId = useSelect(
-		( select ) =>
-			select( 'core/block-editor' ).getPreviousBlockClientId( clientId ),
-		[ clientId ]
-	);
+			blockIndex: getBlockIndex( clientId, rootClientId ),
+		};
+	}, [] );
```

Comment:
Should this have `clientId, rootClientId` as dependencies?

Reply:
Fixed in 0ac181d